drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c

```diff
@@ -2754,6 +2784,8 @@ static int bcm2835_codec_remove(struct platform_device *pdev)
 
 	bcm2835_codec_destroy(drv->decode);
 
+	media_device_unregister(&drv->mdev);
```

Comment:
On the basis that fitting is the reverse of removal, it seems nice to unregister the media device first, but is it necessary?

Reply:
Yes, I think it should happen before in fact, and I thnik I'm also missing calls to v4l2_m2m_unregister_media_controller() and media_device_cleanup().
I'll update this and repost.

Reply:
Updated, and tested with inserting and removing the module. Updated branch to follow